apps/gdalbuildvrt_lib.cpp

```diff
@@ -699,27 +700,27 @@ std::string VRTBuilder::AnalyseRaster(GDALDatasetH hDS,
             maxY = ds_maxY;
         }
 
-        // if not provided an explicit band list, take the one of the first
-        // dataset
-        if (nSelectedBands == 0)
+        if (!bSeparate)
         {
-            nSelectedBands = nTotalBands;
-            CPLFree(panSelectedBandList);
-            panSelectedBandList =
-                static_cast<int *>(CPLMalloc(nSelectedBands * sizeof(int)));
+            // if not provided an explicit band list, take the one of the first
+            // dataset
+            if (nSelectedBands == 0)
```

Comment:
Isn't this code comment now outdated?

Reply:
no, this is in the default/non-separate case